sdk/core/Azure.Core.TestFramework/src/RecordSession.cs

```diff
@@ -79,14 +79,16 @@ public void Record(RecordEntry entry)
             }
         }
 
-        public RecordEntry Lookup(Request request, RecordMatcher matcher, RecordedTestSanitizer sanitizer)
+        public RecordEntry Lookup(Request request, RecordMatcher matcher, RecordedTestSanitizer sanitizer, Func<RecordEntry, bool> skipRequestBodyFilter)
         {
             var requestEntry = RecordTransport.CreateEntry(request, null);
```

Comment:
Let's pull `requestEntry` creating into PlaybackTransport.cs and use the filter to reset body there.

This way we can limit the spread of this flag just to the transports.